buildtest/buildsystem/schemas/script/script-v0.0.1.schema.json

```diff
@@ -0,0 +1,57 @@
+{
+  "$id": "https://HPC-buildtest.github.io/schemas/script/script-v0.0.1.schema.json",
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "BuildTest Test Configuration for Script",
+  "type": "object",
+  "required": ["type", "run"],
+  "propertyNames": {
+    "pattern": "^[A-Za-z_][A-Za-z0-9_]*$"
+  },
+  "properties": {
+    "type": {
+      "type": "string",
+      "pattern": "^script$"
+    },
+    "description": {
+      "type": "string",
+      "description": "A description for the build recipe."
+    },
+    "maintainers": {
+      "type": "array",
+      "description": "One or more maintainers or aliases",
+      "minItems": 1,
+      "items": {
+        "type": "string"
+      }
+    },
+    "env": {
+      "type": "object",
+      "description": "One or more key value pairs for an environment (key=value)",
+      "minItems": 1,
+      "items": {
+        "type": "object",
+        "minItems": 1,
+        "propertyNames": {
+           "pattern": "^[A-Za-z_][A-Za-z0-9_]*$"
+        }
+      }
+    },
+    "shell": {
+      "type": "string",
+      "default": "bash",
+      "description": "The shell interpreter to use."
+    },
+    "pre_run": {
```

Comment:
This is a reasonable request, it will need to go through in the schemas repo first. Why did you approve there if you didn't ultimately agree?

Reply:
I think it's a mistake to remove these, because we need to be able to support the user and test config being able to set precedence for the order of things. For example, a recipe might have the user put commands to do some git or other preparatory steps in pre_run, and then the run command is generated to be some default for the config, and then post_run might be a custom cleanup desired by the user. If we only provide one section, run, the user is going to give a dump of commands and the parser of those lines will have no way to separate them into logical stages. Akin to how spack has clearly defined stages, we at least need simple pre, and post ones here.

Reply:
It also does not hurt to have them, because there is no extra dependency to having them, the user doesn't need to define them, and minimally it just exposes these sections if there is some pre or post command that the user wants to run.